src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -0,0 +1,275 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_integration.h>
+#include <tuple>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+/// Numerical integration algorithms
+namespace integrate {
+
+/// The integration result is returned in this format.
+struct Result {
+  double value;
+  double error;
```

Comment:
@wthrowe here I'd be particularly interested in your opinion. Our current design is to return integration results in this struct and to ignore the GSL functions' `status` return value. Do you have a suggestion how to best handle the integration result, its error and potential failures?

Reply:
For a successful return I think you can either do what you have here or just not return the error estimate.  The error had better be less than the requested tolerance, and in practice that's all people will usually care about.

For the failure cases, I can't think of anything better than throwing an exception.  I don't really like using exceptions, but it does at least make the interface similar to that of the root finders.

Reply:
yes I agree, let's do that. @tomwlodarczyk could you remove the `Result` struct and just return the `value` directly? You can ignore the `error`.
For failures you can include `ErrorHandling/Exceptions.hpp` and `throw convergence_error` as is done in the root finding, e.g. in `RootFinding/NewtonRaphson.hpp`.